using-stm/src/main/java/org/acme/quickstart/stm/FlightResource.java

```diff
@@ -0,0 +1,54 @@
+package org.acme.quickstart.stm;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@Path("/stm")
+@RequestScoped
+public class FlightResource {
+    @Inject
+    FlightServiceFactory factory;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public CompletionStage<String> bookingCount() {
+        return CompletableFuture.supplyAsync(
+                () -> getInfo(factory.getInstance())
```

Comment:
We don't recommend using `supplyAsync` without a specific _executor_. Can you adjust?

Reply:
+1